gguf-py/gguf/gguf_writer.py

```diff
@@ -57,7 +57,7 @@ class GGUFWriter:
 
     def __init__(
         self, path: os.PathLike[str] | str, arch: str, use_temp_file: bool = True,
-        endianess: GGUFEndian = GGUFEndian.LITTLE,
+        endianess: GGUFEndian = GGUFEndian.LITTLE, dont_add_architecture: bool = False
```

Comment:
Maybe rename `dont_add_architecture` to `add_architecture` with a default to `True` instead? Otherwise it seems this is always used with `not`, both when set and when used.

And the double negation can be somewhat confusing.

Reply:
I was thinking about that. Honestly I have no idea why I went with the negative anymore.

Reply:
Resolved with 3328b0a.